index.bs

```diff
@@ -10877,35 +10874,74 @@ defined on that interface.
 These properties are described in more detail
 in sections [[#es-attributes]] and [[#es-operations]].
 
-As with the [=interface object=],
-the [=interface prototype object=] also has properties that
-correspond to the [=constants=] defined on that interface,
-described in [[#es-constants]].
-
-If the [{{NoInterfaceObject}}] extended attribute was not specified on the interface,
-then the [=interface prototype object=] must also
-have a <code class="idl">constructor</code> data property with attributes
-{ \[[Writable]]: <emu-val>true</emu-val>, \[[Enumerable]]: <emu-val>false</emu-val>, \[[Configurable]]: <emu-val>true</emu-val> }
-whose value is a reference to the [=interface object=] for the interface.
-
-<div algorithm="value of the [[Prototype]] internal slot">
+<div algorithm="to create an interface prototype object">
 
-    The [=interface prototype object=]
-    for a given interface |A| must have a
-    \[[Prototype]] [=internal slot=] whose value is returned from
-    the following steps:
+    The [=interface prototype object=] for a given [=interface=] |interface| and [=Realm=] |realm|
+    is created as follows:
 
-    1.  If |A| is declared with the [{{Global}}] [=extended attribute=], and
-        |A| [=support named properties|supports named properties=], then
-        return the [=named properties object=]
-        for |A|, as defined in [[#named-properties-object]].
-    1.  Otherwise, if |A| is declared to inherit from another
-        interface, then return the
-        [=interface prototype object=]
-        for the inherited interface.
-    1.  Otherwise, if |A| is declared with the [{{LegacyArrayClass}}]
-        extended attribute, then return {{%ArrayPrototype%}}.
-    1.  Otherwise, return {{%ObjectPrototype%}}.
+    1.  Let |proto| be null.
+    1.  If |interface| is declared with the [{{Global}}] [=extended attribute=],
+        and |interface| [=support named properties|supports named properties=],
+        then set |proto| to the [=named properties object=] of |interface|,
+        as defined in [[#named-properties-object]].
+    1.  Otherwise, if |interface| is declared to inherit from another interface,
+        then set |proto| to the [=interface prototype object=] of the inherited interface.
+    1.  Otherwise, if |interface| is declared with the [{{LegacyArrayClass}}] [=extended attribute=],
+        then set |proto| to |realm|.\[[Intrinsics]].\[[{{%ArrayPrototype%}}]].
```

Comment:
This becomes rendered as "<var>realm</var>.[[Intrinsics]].\[[%ArrayPrototype%]]"